custom_components/better_thermostat/events/trv.py

```diff
@@ -203,11 +208,17 @@ def update_hvac_action(self):
         self.async_write_ha_state()
         return
 
+    _current_trv_temperature = reduce_attribute(
+        states, ATTR_CURRENT_TEMPERATURE, reduce=lambda *data: min(data)
+    )
+    _current_trv_target_temperature = reduce_attribute(
+        states, set_temperature, reduce=lambda *data: max(data)
+    )
     # return action off if all are off
     if all(a == HVACAction.OFF for a in hvac_actions):
         self.attr_hvac_action = HVACAction.OFF
     # else check if is heating
-    elif self.bt_target_temp > self.cur_temp:
```

Comment:
i think this condition is better. because in the moment when the change happens. we get a new external temperature and BT knows heating is over. also with groups, i don't want the heating idle state to jump around.

Reply:
Traceback (most recent call last):
File "/config/custom_components/better_thermostat/events/trv.py", line 36, in trigger_trv_change
update_hvac_action(self)
File "/config/custom_components/better_thermostat/events/trv.py", line 221, in update_hvac_action
elif _current_trv_target_temperature > _current_trv_temperature:
TypeError: '>' not supported between instances of 'NoneType' and 'float'

the new condition gives me also an error on startup and seems also to happen later, after the restart.

Reply:
sometimes with local_calibration the real current temp is under the real current temp, i try to avoid showing no heating even if it heats this way

Reply:
i'll push a fix for this error

Reply:
I still think it's too buggy.
Let's take an example.

- current trv target temp = 20.0, current trv temp = 19.5, external sensor = 19.5, bt target temp = 20.0
- current trv temp changes to 20.1
- trigger_trv_change is called and calls update_hvac_action
- update_hvac_action checks the temps and the if condition is true because current temp > target temp, so IDLE
- continue trigger_trv_change, external sensor is still below bt target temp, so we adjust the trv target temp to 20.6
- trigger_trv_change is called (because of the trv target temp change), calls again update_hvac_action and sets it back to HEATING (in between is the calculate_heating_power and it's getting messed up)

with the old condition, the hvac_action wouldn't have changed, because external sensor < bt target temp

Reply:
Yes i see the Problem there, i'll change it.

The Problem with local calibration is sometimes too low shoud be fixed in General but this is a other To-Do